### PR TITLE
[csl] allow to configure the CSL auto synchronization

### PR DESCRIPTION
**Pull request by canisLupus1313**
This commit allows configuration of auto synchronisation of CSL,
with data poll mechanism.

### CODE REVIEW

File: src/core/config/mac.h [hunk 1]
```diff
@@ -355,6 +355,16 @@
 #define OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAC_CSL_AUTO_SYNC_ENABLE
+ *
+ * This setting configures CSL auto synchronization based on dtata poll mechanism in Thread 1.2.
```
Comment (edmont) [inline src/core/config/mac.h:361]:
```suggestion
 * This setting configures CSL auto synchronization based on data poll mechanism in Thread 1.2.
```

**Review by edmont — COMMENTED** [review]
Thanks. I believe you should enable this configuration by default, as this is the default CSL receiver behavior right now.

File: src/core/config/mac.h (hunk 1 above)
Comment (canisLupus1313) [reply src/core/config/mac.h]: Fixed.

**Review by edmont — APPROVED** [review]

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -137,6 +137,13 @@ if(OT_CSL_RECEIVER)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE=1")
 endif()
 
+option(OT_CSL_AUTO_SYNC "enable datapolling based on csl config" ${OT_CSL_RECEIVER})
```
Comment (jwhui) [inline etc/cmake/options.cmake:140]:
```suggestion
option(OT_CSL_AUTO_SYNC "enable data polling based on csl config" ${OT_CSL_RECEIVER})
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**